### PR TITLE
REST job class

### PR DESCRIPTION
**Pull request by DarylPenetrante**
### Description
- Introduce a Job class that will use the new Memsource REST API.
- Introduce a `_delete` method that will handle DELETE requests.
- Some Memsource resources returns 204 for a DELETE request which `response.json()` throws an error. Handle this by checking if the status_code is 204 and return an empty dictionary instead. [Requests.json()](https://requests.readthedocs.io/en/master/user/quickstart/#json-response-content)

- For now, we will only support what we already have.

https://github.com/gengo/memsource-wrap/blob/bd6db2e866ae5ca0e66aca5e77ea44af716d9a3a/memsource/api.py#L385-L623

### CODE REVIEW

File: memsource/api_rest/job.py [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+            page: int = 0,
+    ) -> List[models.JobPart]:
+        jobs = self._get("v2/projects/{}/jobs".format(project_id), {"page": page})
+        return [models.JobPart(job_part) for job_part in jobs["content"]]
+
+    def pre_translate(
+            self,
+            project_id: int,
+            job_parts: List[Dict[str, str]],
+            translation_memory_threshold: float=0.7,
```
Comment (roniemartinez) [inline memsource/api_rest/job.py:119]: We could make this constant

File: memsource/api_rest/job.py [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+
+        :param job_uid: job UID.
+        """
+        def get_completed_file_stream() -> types.GeneratorType:
+            return self._get_stream(
+                "v1/projects/{}/jobs/{}/targetFile".format(project_id, job_uid)
+            ).iter_content(constants.CHUNK_SIZE)
+
+        buffer = io.BytesIO()
+        [buffer.write(chunk) for chunk in get_completed_file_stream()]
```
Comment (roniemartinez) [inline memsource/api_rest/job.py:147]: I think we can simplify this without adding an inner function. 👀

File: memsource/constants.py [hunk 3]
```diff
@@ -48,12 +48,24 @@ class HttpMethod(enum.Enum):
     get = 'get'
     post = 'post'
     put = 'put'
+    delete = "delete"
```
Comment (roniemartinez) [inline memsource/constants.py:51]: Hoping we could apply Black here to have more consistent use of quotes.

Comment (DarylPenetrante) [reply memsource/constants.py:51]: Agreed. I will add a ticket to address this. 🙇‍♂️

**Pull request merged**